src/systems/filecoin_markets/storage_market/storage_market_actor_code.go

```diff
@@ -61,25 +50,36 @@ func DeserializeState(x Bytes) State {
 
 ////////////////////////////////////////////////////////////////////////////////
 
-func (a *StorageMarketActorCode_I) WithdrawBalance(rt Runtime, amountRequested actor.TokenAmount) {
-	rt.ValidateImmediateCallerAcceptAnyOfType(actor.BuiltinActorID_Account)
+func _rtGetMinerAccountsAssert(rt Runtime, minerAddr addr.Address) (ownerAddr addr.Address, workerAddr addr.Address) {
+	ownerAddr = addr.Deserialize_Address_Compact_Assert(
+		rt.SendQuery(minerAddr, ai.Method_StorageMinerActor_GetOwnerAddr, []util.Serialization{}))
+
+	workerAddr = addr.Deserialize_Address_Compact_Assert(
+		rt.SendQuery(minerAddr, ai.Method_StorageMinerActor_GetWorkerAddr, []util.Serialization{}))
+
+	return
+}
+
+func (a *StorageMarketActorCode_I) WithdrawBalance(rt Runtime, minerAddr addr.Address, amountRequested actor.TokenAmount) {
+	ownerAddr, workerAddr := _rtGetMinerAccountsAssert(rt, minerAddr)
+	rt.ValidateImmediateCallerInSet([]addr.Address{ownerAddr, workerAddr})
+	// TODO: should workerAddr be permitted here? Depends on actor principal security assumptions.
+
 	callerAddr := rt.ImmediateCaller()
-	TODO() // Restrict caller to miner owner? (May require an additional argument
-	// to specify miner actor.)
 
 	if amountRequested < 0 {
 		rt.AbortArgMsg("sma.WithdrawBalance: negative amount.")
 	}
 
 	h, st := a.State(rt)
 
-	if !st._addressEntryExists(callerAddr) {
+	if !st._addressEntryExists(minerAddr) {
 		rt.AbortArgMsg("sma.WithdrawBalance: address entry does not exist")
 	}
 
-	minBalance := st._getLockedReqBalance(callerAddr)
+	minBalance := st._getLockedReqBalance(minerAddr)
 	newTable, amountExtracted, ok := actor.BalanceTable_WithExtractPartial(
-		st.EscrowTable(), callerAddr, amountRequested, minBalance)
+		st.EscrowTable(), minerAddr, amountRequested, minBalance)
 	Assert(ok)
 	st.Impl().EscrowTable_ = newTable
 
```

Comment:
I don't think that's true (assuming our line indices are aligned) -- line 90 updates the SMA's internal balance table, which doesn't have entries for owners (only for providers and clients). Line 99 sends to `callerAddr` rather than `ownerAddr`, which is a policy question (should worker accounts be allowed to add/withdraw from their corresponding miner's balance, or should only owner accounts be able to do this?)

Reply:
Yes sorry indices are off, i meant line 99. We can settle this in a follow up.